rai/node/rpc.cpp

```diff
@@ -1970,56 +2079,100 @@ void rai::rpc_handler::key_expand ()
 
 void rai::rpc_handler::ledger ()
 {
+	std::error_code ec;
 	if (rpc.config.enable_control)
 	{
 		rai::account start (0);
-		uint64_t count (std::numeric_limits<uint64_t>::max ());
 		boost::optional<std::string> account_text (request.get_optional<std::string> ("account"));
 		if (account_text.is_initialized ())
 		{
-			auto error (start.decode_account (account_text.get ()));
-			if (error)
-			{
-				error_response (response, "Invalid starting account");
-			}
-		}
-		boost::optional<std::string> count_text (request.get_optional<std::string> ("count"));
-		if (count_text.is_initialized ())
-		{
-			auto error_count (decode_unsigned (count_text.get (), count));
-			if (error_count)
+			if (start.decode_account (account_text.get ()))
 			{
-				error_response (response, "Invalid count limit");
+				ec = error_rpc::invalid_starting_account;
 			}
 		}
-		uint64_t modified_since (0);
-		boost::optional<std::string> modified_since_text (request.get_optional<std::string> ("modified_since"));
-		if (modified_since_text.is_initialized ())
+		auto count (decode_unsigned (request.get<std::string> ("count", ""), std::numeric_limits<uint64_t>::max ()));
+		if (!count)
```

Comment:
Here's the condition that should have a `!ec` if we're going with that style.

Reply:
Ah, good catch, I'll add that.

Reply:
Updated